src/main/java/gregtech/common/metatileentities/multi/multiblockpart/MetaTileEntityFluidHatch.java

```diff
@@ -212,92 +219,104 @@ public void registerAbilities(List<IFluidTank> abilityList) {
-            if (translation == null) {
-                translation = GTUtility.getFluidTranslation(this.lockedFluid);
-            }
-
-            if (translation != null) {
-                list.add(translation);
-            }
-        };
+    @Override
+    public ModularPanel buildUI(PosGuiData guiData, PanelSyncManager guiSyncManager) {
+        var fluidSyncHandler = GTFluidSlot.sync(fluidTank)
+                .showAmount(false)
+                .accessibility(false, false)
+                .onLockFluid(stack -> {
+                    if (!isExportHatch) return;
+
+                    if (stack == null) {
+                        this.setLocked(false);
+                        this.lockedFluid = null;
+                    } else {
+                        this.setLocked(true);
+                        this.lockedFluid = stack.copy();
+                        this.lockedFluid.amount = 1;
+                    }
+                });
+
+        return GTGuis.createPanel(this, 176, 166)
+                .child(IKey.lang(getMetaFullName()).asWidget().pos(6, 6))
+
+                // export specific
+                .childIf(isExportHatch, new ItemSlot()
+                        .pos(90, 44)
+                        .background(GTGuiTextures.SLOT, GTGuiTextures.OUT_SLOT_OVERLAY)
+                        .slot(new ModularSlot(exportItems, 0)
+                                .accessibility(false, true)))
+                .childIf(isExportHatch, new ToggleButton()
+                        .pos(7, 63)
+                        // todo lock overlay
+                        .value(new BooleanSyncValue(this::isLocked, this::setLocked))
+                        // todo tooltip is not working for some reason, fixed in rc3
+                        .tooltip(t -> t.addLine(IKey.lang("gregtech.gui.fluid_lock.tooltip"))))
+
+                // import specific
+                .childIf(!isExportHatch, GTGuiTextures.TANK_ICON.asWidget()
+                        .pos(91, 36)
+                        .size(14, 15))
+                .childIf(!isExportHatch, new ItemSlot()
+                        .pos(90, 53)
+                        .background(GTGuiTextures.SLOT, GTGuiTextures.OUT_SLOT_OVERLAY)
+                        .slot(new ModularSlot(exportItems, 0)
+                                .accessibility(false, true)))
+
+                // common ui
+                .child(Flow.column()
+                        .crossAxisAlignment(Alignment.CrossAxis.START)
+                        .size(81, isExportHatch ? 46 : 55)
+                        .padding(3, 4)
+                        .background(GTGuiTextures.DISPLAY)
+                        .pos(7, 16)
+                        .child(IKey.lang("gregtech.gui.fluid_amount").asWidget()
+                                .widthRel(1f)
+                                .height(10))
+                        // IKey's do not like empty strings...
+                        .child(IKey.dynamic(getFluidName(fluidSyncHandler))
```

Comment:
These name and amount text widgets could use some color for the text, right now they are a little hard to see against the background.